src/native/libs/System.Security.Cryptography.Native.Apple/CMakeLists.txt

```diff
@@ -38,6 +39,31 @@ else()
     )
 endif()
 
+# As of CMake 3.20.2 support for Swift only works with the Ninja and XCode
+# generators so we cannot rely on it. Even with the Ninja generator it doesn't
+# work in combination with other languages within the same library.
+if (NOT SWIFT_COMPILER_TARGET AND CLR_CMAKE_TARGET_OSX)
+    set(SWIFT_PLATFORM "macosx")
+    set(SWIFT_PLATFORM_SUFFIX "")
+    set(SWIFT_DEFAULT_DEPLOYMENT_TARGET "10.15")
+
+   if(CMAKE_OSX_DEPLOYMENT_TARGET)
+     set(SWIFT_DEPLOYMENT_TARGET ${CMAKE_OSX_DEPLOYMENT_TARGET})
+   else()
+     set(SWIFT_DEPLOYMENT_TARGET ${SWIFT_DEFAULT_DEPLOYMENT_TARGET})
+   endif()
+
+   set(SWIFT_COMPILER_TARGET "${CMAKE_OSX_ARCHITECTURES}-apple-${SWIFT_PLATFORM}${SWIFT_DEPLOYMENT_TARGET}${SWIFT_PLATFORM_SUFFIX}")
+endif()
+
+add_custom_command(
+    OUTPUT pal_aead.o
+    COMMAND xcrun swiftc -emit-object -static -parse-as-library -runtime-compatibility-version none -sdk ${CMAKE_OSX_SYSROOT} -target ${SWIFT_COMPILER_TARGET} ${CMAKE_CURRENT_SOURCE_DIR}/pal_aead.swift -o pal_aead.o
+    MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/pal_aead.swift
+    COMMENT "Compiling Swift file pal_aead.swift"
+)
```

Comment:
If it's going to take this much content in the makefile for every swift translator/forwarder we add, then we might want to aim generically here, and just call it something like `swiftbindings.swift`.